docs/src/main/asciidoc/reactive-postgres-client.adoc

```diff
@@ -0,0 +1,450 @@
+That's it!
+So far we have seen how to configure a pooled client and execute simple queries.
+We are now ready to develop the data management code and implement our RESTful endpoint.
+
+== Using
+
+=== Query results traversal
+
+In development mode, the database is set up with a few rows in the `fruits` table.
+To retrieve all the data, we will use the `query` method again:
+
+[source,java]
+----
+CompletionStage<PgRowSet> rowSet = client.query("SELECT id, name FROM fruits ORDER BY name ASC");
+----
+
+When the operation completes, we will get a `PgRowSet` that has all the rows buffered in memory.
+It can be traversed with an `Iterator`:
+
+[source,java]
+----
+CompletionStage<List<Fruit>> fruits = rowSet.thenApply(pgRowSet -> {
+    List<Fruit> list = new ArrayList<>(pgRowSet.size());
+    PgIterator pgIterator = pgRowSet.iterator();
+    while (pgIterator.hasNext()) {
+        list.add(from(pgIterator.next()));
+    }
+    return list;
+});
+----
+
+The `from` method converts a `Row` instance to a `Fruit` instance.
+It is extracted as a convenience for the implementation of the other data management methods:
+
+[source,java]
+.src/main/java/org/acme/vertx/Fruit.java
+----
+private static Fruit from(Row row) {
+    return new Fruit(row.getLong("id"), row.getString("name"));
+}
+----
+
+Putting it all together, the `Fruit.findAll` method looks like:
+
+[source,java]
+.src/main/java/org/acme/vertx/Fruit.java
+----
+    public static CompletionStage<List<Fruit>> findAll(PgPool client) {
+        return client.query("SELECT id, name FROM fruits ORDER BY name ASC").thenApply(pgRowSet -> {
+            List<Fruit> list = new ArrayList<>(pgRowSet.size());
+            PgIterator pgIterator = pgRowSet.iterator();
+            while (pgIterator.hasNext()) {
+                list.add(from(pgIterator.next()));
+            }
+            return list;
+        });
+    }
+----
+
+And the endpoint to get all fruits from the backend:
+
+[source,java]
+.src/main/java/org/acme/vertx/FruitResource.java
+----
+@GET
+public CompletionStage<Response> get() {
+    return Fruit.findAll(client)
+            .thenApply(fruits -> fruits.isEmpty() ? Response.noContent() : Response.ok(fruits))
+            .thenApply(ResponseBuilder::build);
+}
+----
+
+Now start {project-name} in `dev` mode with:
+
+[source, shell]
+----
+mvn compile quarkus:dev
+----
+
+Eventually, open your browser and navigate to http://localhost:8080/fruits, you should see:
+
+[source, shell]
+----
+[{"id":3,"name":"Apple"},{"id":1,"name":"Orange"},{"id":2,"name":"Pear"}]
+----
+
+=== Prepared queries
+
+The Reactive Postgres Client can also prepare queries and take parameters that are replaced in the SQL statement at execution time:
+
+[source,java]
+----
+client.preparedQuery("SELECT name FROM fruits WHERE id = $1", Tuple.of(id))
+----
+
+TIP: The SQL string can refer to parameters by position, using $1, $2, ...etc.
+
+As the simple `query` method, `preparedQuery` eventually returns a `PgRowSet`.
```

Comment:
s/As/Like/

Reply:
I think you should drop `eventually` here. I suspect it implies that it returns a `CS<PgRowSet>` but I don't think `eventually` is the industry term for promises ;)